q2cli/handlers.py

```diff
@@ -310,12 +310,25 @@ def get_click_options(self):
 
     def get_value(self, arguments, fallback=None):
         import qiime2
-
+        import qiime2.sdk.util
         path = self._locate_value(arguments, fallback)
+
         if path is None:
             return None
+
+        try:
+            # check to see if path is an artifact
+            art = qiime2.Artifact.load(path)
+        except Exception:
+            pass
         else:
-            return qiime2.Metadata.load(path)
+            if qiime2.sdk.util.has_metadata(art):
```

Comment:
How about moving `.has_metadata()` to a method on the `Artifact` class? It'd make it easier to discover instead of being orphaned in `sdk.util`.